src/App.js

```diff
@@ -166,26 +83,16 @@ export default class App extends React.Component<{}, AppState> {
         }
     }
 
-    handleChangeLanguage = (event: SyntheticEvent<HTMLSelectElement>) => {
-        this.setStateSettings({
-            language: event.currentTarget.value
-        });
-    };
-
     handleChangeProgram = (program: Program) => {
         this.setState({
             program: program
         });
     };
 
     handleClickRun = () => {
-        if (this.turtleGraphicsRef.current !== null) {
-            this.turtleGraphicsRef.current.clear();
-        }
-        if (this.turtleGraphicsRef.current !== null) {
-            this.turtleGraphicsRef.current.home();
+        if (this.state.dashConnectionStatus === 'connected') {
+            this.interpreter.run(this.state.program);
```

Comment:
As discussed on IRC, let's instead disable the "Run" button when dashConnectionStatus !== 'connected'